From 716bfe261a7d0a7a01d78d58b6d2d0ccf7ce7649 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Mon, 10 Apr 2017 13:50:51 +0800
Subject: [PATCH 1658/5242] MLK-14655-1 video: mxsfb: don't clear AS regs when
 overlay fb release

commit  7268002c8038876a3df8cad3e0821109fffb2060 from
https://source.codeaurora.org/external/imx/linux-imx.git

Don't clear the 'LCDC_AS_CTRL' and 'LCDC_AS_NEXT_BUF' registers
in the function 'overlayfb_release()', since the next user may
enable overlay fb without calling set_par first.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
(cherry picked from commit 167b1f430d2bb178d22f32f790d55b08352a6a87)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxsfb.c |    3 ---
 1 file changed, 3 deletions(-)

diff --git a/drivers/video/fbdev/mxsfb.c b/drivers/video/fbdev/mxsfb.c
index 4dd0d83a..3c9d383 100644
--- a/drivers/video/fbdev/mxsfb.c
+++ b/drivers/video/fbdev/mxsfb.c
@@ -1716,7 +1716,6 @@ static int overlayfb_open(struct fb_info *info, int user)
 static int overlayfb_release(struct fb_info *info, int user)
 {
 	struct mxsfb_layer *ofb = (struct mxsfb_layer*)info->par;
-	struct mxsfb_info *fbi = ofb->fbi;
 
 	BUG_ON(!atomic_read(&ofb->usage));
 
@@ -1725,8 +1724,6 @@ static int overlayfb_release(struct fb_info *info, int user)
 			ofb->ops->disable(ofb);
 
 		ofb->blank_state = -1;
-		writel(0x0, fbi->base + LCDC_AS_CTRL);
-		writel(0x0, fbi->base + LCDC_AS_NEXT_BUF);
 	}
 
 	return 0;
-- 
1.7.9.5

